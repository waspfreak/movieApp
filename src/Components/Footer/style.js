import styled from 'styled-components';

export const StyledFooter = styled.div`
	.footer {
		font-family: 'Titillium Web', sans-serif;
		overflow: hidden;
		background: #2b2b2a;
		font-size: 0.95rem;
		color: #fff;
		text-shadow: 0 0.1rem 0.1rem rgba(0, 0, 0, 0.25);
		padding-left: 20px;
		padding-top: 36px;
		.link {
			color: #fff;
		}
	}
	@media screen and (max-width: 500px) {
		opacity: 0.9;
		font-size: 0.9rem;
	}
`;

export const StyledTMDBLogo = styled.img`
	width: 122px;
	float: right;
	margin: 0 1rem 0.5rem;
	filter: opacity(0.9) brightness(2);

	@media screen and (max-width: 500px) {
		display: inline-block;
		width: 90px;
		margin: 0 10px;
	}
`;
