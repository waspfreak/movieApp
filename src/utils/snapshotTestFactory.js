import {cleanup, render} from '@testing-library/react';
import React from 'react';

/**
 *
 * Sets up snapshot tests in a standard way and avoids repeating boiler plate
 */
export function snapshotTestFactory(Component: any, properties: any) {
	/**
	 * Re-usable it block
	 */
	const itBlock = (props: any) => {
		it('should match snapshot', () => {
			const {container} = render(<Component {...props} />);
			expect(container).toMatchSnapshot();
		});
	};

	/**
	 * Re-usable describe block
	 */
	describe(`${Component.name}`, () => {
		/**
		 * Clean up any mock functions that might be in props
		 */
		afterEach(() => {
			jest.clearAllMocks();
			cleanup();
		});

		/**
		 * If its an array of props then iterate them into seperate it blocks
		 */
		if (Array.isArray(properties)) {
			properties.forEach((props) => {
				itBlock(props);
			});
		} else {
			itBlock(properties);
		}
	});
}
