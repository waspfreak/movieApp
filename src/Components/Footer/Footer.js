import React from 'react';
// Styled Components
import {StyledFooter, StyledTMDBLogo} from './style';

import TMDBLogo from '../../assets/tmdb_logo.svg';

export const Footer = () => {
	return (
		<StyledFooter>
			<div className='footer'>
				<a className='link' href='https://www.themoviedb.org/'>
					<StyledTMDBLogo src={TMDBLogo} alt='tmdb-logo' target='_blank' />
				</a>
				Crafted by{' '}
				<a className='link' href='https://github.com/waspfreak' target='_blank'>
					Juliana Leon
				</a>{' '}
				with{' '}
				<a className='link' href='https://reactjs.org/' target='_blank'>
					{' '}
					React
				</a>
			</div>
		</StyledFooter>
	);
};

export default Footer;
