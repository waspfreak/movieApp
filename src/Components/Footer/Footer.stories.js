import React from 'react';

import {Footer} from './Footer';

export default {
	title: 'Example/Footer',
	component: Footer,
};

const Template = (args) => <Footer {...args} />;

export const FooterExample = Template.bind({});
FooterExample.args = {};
