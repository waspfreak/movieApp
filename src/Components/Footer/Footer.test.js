import {snapshotTestFactory} from '../../utils/snapshotTestFactory';
//
import {Footer} from './Footer';
const props: Footer[] = [
	{
		posterPath:
			'http://image.tmdb.org/t/p/w500/lPsD10PP4rgUGiGR4CCXA6iY0QQ.jpg',
		originalTitle: 'Movie Title',
	},
];
snapshotTestFactory(Footer, props);
