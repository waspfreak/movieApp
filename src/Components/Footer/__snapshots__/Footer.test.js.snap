// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer should match snapshot 1`] = `
.c0 .footer {
  font-family: 'Titillium Web',sans-serif;
  overflow: hidden;
  background: #2b2b2a;
  font-size: 0.95rem;
  color: #fff;
  text-shadow: 0 0.1rem 0.1rem rgba(0,0,0,0.25);
  padding-left: 20px;
  padding-top: 36px;
}

.c0 .footer .link {
  color: #fff;
}

.c1 {
  width: 122px;
  float: right;
  margin: 0 1rem 0.5rem;
  -webkit-filter: opacity(0.9) brightness(2);
  filter: opacity(0.9) brightness(2);
}

@media screen and (max-width:500px) {
  .c0 {
    opacity: 0.9;
    font-size: 0.9rem;
  }
}

@media screen and (max-width:500px) {
  .c1 {
    display: inline-block;
    width: 90px;
    margin: 0 10px;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="footer"
    >
      <a
        class="link"
        href="https://www.themoviedb.org/"
      >
        <img
          alt="tmdb-logo"
          class="c1"
          src="tmdb_logo.svg"
          target="_blank"
        />
      </a>
      Crafted by
       
      <a
        class="link"
        href="https://github.com/waspfreak"
        target="_blank"
      >
        Juliana Leon
      </a>
       
      with
       
      <a
        class="link"
        href="https://reactjs.org/"
        target="_blank"
      >
         
        React
      </a>
    </div>
  </div>
</div>
`;
